gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorage.java

```diff
@@ -193,11 +196,38 @@ SeekableByteChannel open(StorageResourceId resourceId, GoogleCloudStorageReadOpt
    *     exist
    * @throws IOException in all other error cases
    */
-  void copy(
+  default void copy(
       String srcBucketName,
       List<String> srcObjectNames,
       String dstBucketName,
       List<String> dstObjectNames)
+      throws IOException {
+    checkArgument(srcObjectNames != null, "srcObjectNames must not be null");
+    checkArgument(dstObjectNames != null, "dstObjectNames must not be null");
+
+    List<StorageResourceId> srcObjects =
+        srcObjectNames.stream()
+            .map(srcObjectName -> new StorageResourceId(srcBucketName, srcObjectName))
+            .collect(Collectors.toList());
+    List<StorageResourceId> dstObjects =
+        dstObjectNames.stream()
+            .map(dstObjectName -> new StorageResourceId(dstBucketName, dstObjectName))
+            .collect(Collectors.toList());
+    copy(srcObjects, dstObjects);
+  }
+
+  /**
+   * Copies metadata of the given objects. After the copy is successfully complete, each object blob
+   * is reachable by two different names. Copying between two different locations or between two
+   * different storage classes is not allowed.
+   *
+   * @param srcObjects list of the objects to copy
+   * @param dstObjects list of the objects after copy
+   * @throws java.io.FileNotFoundException if the source object or the destination bucket does not
+   *     exist
+   * @throws IOException in all other error cases
+   */
+  void copy(List<StorageResourceId> srcObjects, List<StorageResourceId> dstObjects)
```

Comment:
We need default implementation for this method to preserve backward compatibility.